File: c/cz/mg/vulkan/PFN_vkCreateObjectTableNVX.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCreateObjectTableNVX_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pObjectTable, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateObjectTableNVX f = (PFN_vkCreateObjectTableNVX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkObjectTableCreateInfoNVX*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkObjectTableNVX*)jniLongToPointer(pObjectTable))
    );
}

File: cz/mg/vulkan/PFNvkCreateObjectTableNVX.java
package cz.mg.vulkan;

public class PFNvkCreateObjectTableNVX extends VkFunctionPointer {
    public PFNvkCreateObjectTableNVX() {
    }

    public PFNvkCreateObjectTableNVX(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreateObjectTableNVX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCreateObjectTableNVX(long value) {
        setValue(value);
    }

    public PFNvkCreateObjectTableNVX(VkInstance instance) {
        super(instance, new VkString("vkCreateObjectTableNVX"));
    }

    public void call(VkDevice device, VkObjectTableCreateInfoNVX pCreateInfo, VkAllocationCallbacks pAllocator, VkObjectTableNVX pObjectTable, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getSinkAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pObjectTable != null ? pObjectTable.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pObjectTable, long rval);
}
